modules/documentObserver.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+var EXPORTED_SYMBOLS = ['OnNewDocument'];
+
+var Cu = Components.utils;
+
+Cu.import('resource://gre/modules/Services.jsm');
+Cu.import('chrome://greasemonkey-modules/content/util.js');
+
+
+var callbacks = new WeakMap()
+
+function OnNewDocument(topWindow, callback) {
+	callbacks.set(topWindow, callback)
```

Comment:
This is another memory leak of every window ever created being stored permanently?